libclamav_rust/src/cdiff.rs

```diff
@@ -612,176 +724,213 @@ fn cmd_move(ctx: &mut Context, move_op: MoveOp) -> Result<(), CdiffError> {
 
     // Test for move with open db
     if let Some(x) = &ctx.open_db {
-        return Err(CdiffError::NotClosedBeforeAction(x.into()));
+        return Err(ProcessingError::NotClosedBeforeAction(x.into()).into());
     }
 
-    // Open src in read-only mode
-    let src_file = File::open(move_op.src)?;
-
     // Open dst in append mode
-    let mut dst_file = OpenOptions::new().append(true).open(move_op.dst)?;
+    let mut dst_file = OpenOptions::new()
+        .append(true)
+        .open(&move_op.dst)
+        .map_err(|e| InputError::Processing(e.into()))?;
 
     // Create tmp file and open for writing
     let tmp_named_file = tempfile::Builder::new()
         .prefix("_tmp_move_file")
-        .tempfile_in("./")?;
+        .tempfile_in("./")
+        .map_err(|e| InputError::Processing(e.into()))?;
     let mut tmp_file = tmp_named_file.as_file();
 
-    // Create a buffered reader and loop over src, line by line
-    let src_reader = BufReader::new(src_file);
-
-    for (mut line_no, line) in src_reader.lines().enumerate() {
-        let line = line?;
-
-        // cdiff files start at line 1
-        line_no += 1;
-        if state == State::Init && line_no == move_op.start_line_no {
-            if line.starts_with(move_op.start_line) {
-                state = State::Move;
-                dst_file.write_all(line.as_bytes())?;
-                dst_file.write_all(b"\n")?;
-            } else {
-                error!("{} does not match {}", line, move_op.start_line);
-                return Err(CdiffError::PatternDoesNotMatch(
-                    "MOVE",
-                    line_no,
-                    move_op.src.to_string(),
-                ));
+    {
```

Comment:
It bounds the lifetime of `src_reader` so that it closes at the end. Otherwise, Windows complains about the open file.

I can also unwrap it and add a `drop(src_reader)` where the block would have ended so that it's more obvious. I'll do that.